dashboard/test/models/pd/teacher_application_test.rb

```diff
@@ -0,0 +1,53 @@
+require 'test_helper'
+
+class Pd::TeacherApplicationTest < ActiveSupport::TestCase
+  test 'required field validations' do
+    teacher_application = Pd::TeacherApplication.new
+    refute teacher_application.valid?
+    assert_equal [
+      'User is required',
+      'Primary email is required',
+      'Secondary email is required',
+      'Application is required'
+    ], teacher_application.errors.full_messages
+
+    teacher_application.user = create :teacher
+    teacher_application.primary_email = 'teacher@example.net'
+    teacher_application.secondary_email = 'teacher@my.school.edu'
```

Comment:
Can you try something like teacher+foo@my.school.edu? Just to better test the validation